lms/product/plugin/course_copy.py

```diff
@@ -0,0 +1,74 @@
+from lms.models import File
+from lms.services.exceptions import CanvasFileNotFoundInCourse
+
+
+class CourseCopyPlugin:
+    file_type = None
+
+    def assert_file_in_course(self, course_id, file_id):
+        """Raise if the current user can't see file_id in course_id."""
+        raise NotImplementedError()
+
+    def find_matching_file_in_course(self, original_file_id, new_course_id):
+        try:
+            # Get the current (copied) courses files, that will have the side effect of storing files in the DB
+            _ = self._store_new_course_files(new_course_id)
+        except ExternalRequestError:
+            # We might not have access to use the API for that endpoint.
+            # That will depend on our role and the course's permissions settings.
+            # We will continue anyway, maybe the files of the new course are already in the DB
+            # after an instructor launched corrected the issue.
+            pass
```

Comment:
This is an improvement over the Canvas solution, store the files in the DB and use those to do the course copy matching.

That means that any launch after the "storing" takes place can in theory fix course copy itself based on the info in the DB.

In canvas we rely on the API call to do so (even if we also store he files in the DB).